openmls/src/group/mls_group/config.rs

```diff
@@ -50,65 +52,155 @@ pub struct MlsGroupConfig {
     pub(crate) number_of_resumption_psks: usize,
     /// Flag to indicate the Ratchet Tree Extension should be used
     pub(crate) use_ratchet_tree_extension: bool,
+    /// Sender ratchet configuration
+    pub(crate) sender_ratchet_configuration: SenderRatchetConfiguration,
+}
+
+impl MlsGroupConfig {
+    /// Returns a builder for [`MlsGroupConfig`].
+    pub fn builder() -> MlsGroupConfigBuilder {
+        MlsGroupConfigBuilder::new()
+    }
+
+    /// Returns the wire format policy set in this  [`MlsGroupConfig`].
+    pub fn wire_format_policy(&self) -> WireFormatPolicy {
+        self.wire_format_policy
+    }
+
+    /// Returns the padding size set in this  [`MlsGroupConfig`].
+    pub fn padding_size(&self) -> usize {
+        self.padding_size
+    }
+
+    /// Returns the [`SenderRatchetConfiguration`] set in this  [`MlsGroupConfig`].
+    pub fn sender_ratchet_configuration(&self) -> &SenderRatchetConfiguration {
+        &self.sender_ratchet_configuration
+    }
+}
+
+/// Specifies configuration pattern for an [`MlsGroup`]. Refer to the [User
+/// Manual](https://openmls.tech/book/user_manual/group_config.html) for more
+/// information about the different configuration values.
+#[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+pub struct MlsGroupPattern {
```

Comment:
I agree with Jan that the naming may not be the best.
But I also wonder if we need both of these.
My understanding is that `Config` is for runtime parameters and `Pattern` is for creating a new group.
How about having one config with different ways of instantiating it depending on whether it's used for join or creation?